stingray/fourier.py

```diff
@@ -0,0 +1,1286 @@
+
+    Examples
+    --------
+    >>> mean = var = 1000000
+    >>> N = 1000000
+    >>> dt = 0.2
+    >>> meanrate = mean / dt
+    >>> lc = np.random.poisson(mean, N)
+    >>> pds = np.abs(fft(lc))**2
+    >>> pdsnorm = normalize_frac(pds, dt, lc.size, mean)
+    >>> np.isclose(pdsnorm[1:N//2].mean(), poisson_level(meanrate=meanrate,norm="frac"), rtol=0.01)
+    True
+    """
+    #     (mean * N) / (mean /dt) = N * dt
+    #     It's Leahy / meanrate;
+    #     Nph = mean * N
+    #     meanrate = mean / dt
+    #     norm = 2 / (Nph * meanrate) = 2 * dt / (mean**2 * N)
+
+    return power * 2. * dt / (mean ** 2 * N)
+
+
+def normalize_abs(power, dt, N):
+    """Absolute rms normalization, from the variance of the lc.
+
+    Examples
+    --------
+    >>> mean = var = 100000
+    >>> N = 1000000
+    >>> dt = 0.2
+    >>> meanrate = mean / dt
+    >>> lc = np.random.poisson(mean, N)
+    >>> pds = np.abs(fft(lc))**2
+    >>> pdsnorm = normalize_abs(pds, dt, lc.size)
+    >>> np.isclose(pdsnorm[1:N//2].mean(), poisson_level(meanrate=meanrate, norm="abs"), rtol=0.01)
+    True
+    """
+    #     It's frac * meanrate**2; Leahy / meanrate * meanrate**2
+    #     Nph = mean * N
+    #     meanrate = mean / dt
+    #     norm = 2 / (Nph * meanrate) * meanrate**2 = 2 * dt / (mean**2 * N) * mean**2 / dt**2
+
+    return power * 2. / N / dt
+
+
+def normalize_leahy_from_variance(power, variance, N):
+    """Leahy+83 normalization, from the variance of the lc.
+
+    Examples
+    --------
+    >>> mean = var = 100000.
+    >>> N = 1000000
+    >>> lc = np.random.poisson(mean, N).astype(float)
+    >>> pds = np.abs(fft(lc))**2
+    >>> pdsnorm = normalize_leahy_from_variance(pds, var, lc.size)
+    >>> np.isclose(pdsnorm[0], 2 * np.sum(lc), rtol=0.01)
+    True
+    >>> np.isclose(pdsnorm[1:N//2].mean(), poisson_level(norm="leahy"), rtol=0.01)
+    True
+    """
+    return power * 2. / (variance * N)
+
+
+def normalize_leahy_poisson(power, Nph):
+    """Leahy+83 normalization, from the variance of the lc.
+
+    Examples
+    --------
+    >>> mean = var = 100000.
+    >>> N = 1000000
+    >>> lc = np.random.poisson(mean, N).astype(float)
+    >>> pds = np.abs(fft(lc))**2
+    >>> pdsnorm = normalize_leahy_poisson(pds, np.sum(lc))
+    >>> np.isclose(pdsnorm[0], 2 * np.sum(lc), rtol=0.01)
+    True
+    >>> np.isclose(pdsnorm[1:N//2].mean(), poisson_level(norm="leahy"), rtol=0.01)
+    True
+    """
+    return power * 2. / Nph
+
+
+def normalize_periodograms(unnorm_power, dt, N, mean, variance=None, norm="abs", power_type="all"):
+    """Wrapper around all the normalize_NORM methods.
+
+    Normalize the real part of the cross spectrum to Leahy, absolute rms^2,
+    fractional rms^2 normalization, or not at all.
+
+    Parameters
+    ----------
+    unnorm_power: numpy.ndarray
+        The unnormalized cross spectrum.
+
+    dt: float
+        The sampling time of the light curve
+
+    mean: float
```

Comment:
We might not want to call this just `mean`, because if someone imports `from numpy import *`, they might find this super confusing?

Reply:
Good point.